smart-send-logistics/includes/frontend/class-ss-shipping-endpoint.php

```diff
@@ -0,0 +1,126 @@
+    public function get_pickup_points_callback(WP_REST_Request $request)
+    {
+        $country = wc_clean($request->get_param('country'));
+        $postal_code = wc_clean($request->get_param('postCode'));
+        $city = (!empty($request->get_param('city')) ? wc_clean($request->get_param('city')) : null);
+        $street = wc_clean($request->get_param('street'));
+        $carrier = wc_clean($request->get_param('carrier'));
+        $is_rest_api = true;
+
+        $ss_agents = $this->frontend->handle_agents_session($carrier, $country, $postal_code, $city, $street, $is_rest_api);
+        if (!empty($ss_agents)) {
+            $ss_agent_options = array();
+            $ss_setting = SS_SHIPPING_WC()->get_ss_shipping_settings();
+
+            if (!isset($ss_setting['default_select_agent']) || $ss_setting['default_select_agent'] == 'no') {
+                $ss_agent_options[0] = __('- Select Pick-up Point -', 'smart-send-logistics');
+            }
+            foreach ($ss_agents as $key => $agent) {
+                $formatted_address = $this->frontend->get_formatted_address_for_endpoint($agent);
+                $ss_agent_options[$agent->agent_no] = $formatted_address;
+            }
+            return new WP_REST_Response($ss_agent_options, 200);
+        } else {
+            return new WP_REST_Response(array('message' => __('No pick-up points found', 'smart-send-logistics')), 404);
+        }
+    }
+
+    /**
+     * Get shipping method meta data
+     */
+    public function get_shipping_carrier(WP_REST_Request $request)
+    {
+        // Get the shipping method value from the request
+        $shipping_method = wc_clean($request->get_param('ss_method'));
+        if (empty($shipping_method)) {
+            return new WP_REST_Response(array('message' => __('Invalid Shipping Method ID', 'smart-send-logistics')), 400);
+        }
+
+        $carrier_keys = ["name", "id"];
+
+        $shipping_method_parts = explode(":", $shipping_method);
+
+        if (count($shipping_method_parts) > 1) {
+            // Get the second part of the shipping method which is the ID
+            $shipping_method_parts = array_combine($carrier_keys, $shipping_method_parts);
+        }
+
+        $shipping_method_id = $shipping_method_parts["id"];
+
+        // Retrieve the shipping method instance by its ID
+        $shipping_method_instance = WC_Shipping_Zones::get_shipping_method($shipping_method_id);
+        if (!$shipping_method_instance) {
+            return new WP_REST_Response(array('message' => __('No Shipping Method found', 'smart-send-logistics')), 404);
+        }
+        $options = get_option('woocommerce_' . $shipping_method_instance->id . '_' . $shipping_method_instance->instance_id . '_settings');
+        $shipping_agent = $options['method'];
+        $shipping_agent = explode("_", $shipping_agent);
+        $shipping_carrier = array('carrier' => $shipping_agent[0]);
+        return new WP_REST_Response($shipping_carrier, 200);
+    }
```

Comment:
Why is this function needed? Surely the code of the shipping method is already available in then frontend?

Reply:
yes it is used to get the carrier details and it's response is formatted in the recommended format now.